### DISCUSSION TITLE
Inconsistencies in the Distraction Free Settings: Hide Videos From Channels

### DISCUSSION THREAD

**Discussion by efb4f5ff-1298-471a-8973-3d47447115dc** [FreeTubeApp/FreeTube#3744] [General]
Im a confused by this feature at this point. The description of the feature says that it should **hide all videos, playlists and the channel itself** from **search** and **trending**.

Videos, playlists and channels are indeed hidden on search and trending **but videos and playlists are also hidden when u go to the channel page itself.**

If this feature should hide everything from the channel everywhere (like it is doing now) then:

- It should also do that on the most popular tab
- It should also hide the videos from channel X that's in a playlist but the playlist is created by channel Y
- It should also do that on the community tab
- It should also hide a channel in the featured channel section in the about tab on the channel page

Otherwise we should do:
- Relist the videos on the channel page
- Hide videos from channels in the recommended section (see explanation about this below)

Channels are listed under the recommended section while the original PR said they would be hidden

> Videos and playlists from this channel ( and the channel itself ) will no longer show up in a search, recommended or trending.

Retrieved from #2849 PR Body

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@PikachuEXE Thanks for addressing both issues with the implementation of this feature. As I was reading over this again i started wondering about one thing, this feature does hide channels from Trending but not from the Most Popular tab. Do you think it would make sense to also hide it from the Most Popular tab? IMO it would make this feature more consistent.

**Reply by PikachuEXE** [reply]
I think it should
Though I forgot to do that in those PRs I created ._.

**Reply by PikachuEXE** [reply]
What is `shortVideo` type...
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/5704d7a7-c484-47d1-8e44-d5a6dc0eb976)

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
> I think it should
Though I forgot to do that in those PRs I created ._.

U can add it in one of the PRs if u want

> What is `shortVideo` type... 

Hmm idk but i have seen it before in a recent PR

**Reply by PikachuEXE** [reply]
Seems like IV specific stuff...
https://docs.invidious.io/api/#get-apiv1popular
![image](https://github.com/FreeTubeApp/FreeTube/assets/1018543/a4b1f4bf-bb8d-4656-9f4a-a0bf07ccc219)

**Reply by PikachuEXE** [reply]
Fixed in https://github.com/FreeTubeApp/FreeTube/pull/3749

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
All inconsistencies are addressed, thanks @PikachuEXE :)
